src/data/set/basic.lean

```diff
@@ -1782,6 +1782,14 @@ lemma injective.nonempty_apply_iff {f : set α → set β} (hf : injective f)
   (h2 : f ∅ = ∅) {s : set α} : (f s).nonempty ↔ s.nonempty :=
 by rw [← ne_empty_iff_nonempty, ← h2, ← ne_empty_iff_nonempty, hf.ne_iff]
 
+lemma injective.mem_range_iff_exists_unique (hf : injective f) {b : β} :
+  b ∈ range f ↔ ∃! a, f a = b :=
+⟨λ ⟨a, h⟩, ⟨a, h, λ a' ha, hf (ha.trans h.symm)⟩, exists_unique.exists⟩
+
+lemma injective.exists_unique_of_mem_range (hf : injective f) {b : β} (hb : b ∈ range f) :
+  ∃! a, f a = b :=
+hf.mem_range_iff_exists_unique.mp hb
```

Comment:
I think this overlaps a lot with the soon-to-be-bors'd `bijective_iff_exists_unique`. Perhaps its worth waiting for that to be merged and see if one proof or the other can be simplified.

Reply:
What's the PR for that?

Reply:
#5995

Reply:
Since `bijective_iff_exists_unique` is up the import hierarchy, it's not possible to use the set lemma to prove it. And the other direction isn't possible either because we don't have the promise that `f` is surjective. So I think it's fine to leave the proof bodies unrelated.